src/containers/Dashboard.js

```diff
@@ -0,0 +1,135 @@
+            hoverBackgroundColor: ['#FF6384', '#13bdd8', '#FFCE56'],
+          },
+        ],
+      },
+      display: Display.FRONTEND,
+    };
+    this.generateLocalData = this.generateLocalData.bind(this);
+    this.getRemoteData = this.getRemoteData.bind(this);
+    this.fetchFromStorageData = this.fetchFromStorageData.bind(this);
+    this.initStorageData();
+  }
+
+  initStorageData() {
+    setStorageData(this.state.chart_data.datasets[0].data);
+  }
+
+  render() {
+    return (
+      <Container>
+        <Row className='justify-content-center mt-5'>
+          <h1>Dashboard App</h1>
+        </Row>
+        <Row className='justify-content-center mb-5 mt-5'>
+          <Button
+            variant='info'
+            size='sm'
+            onClick={this.generateLocalData}
+            style={{ margin: 10 }}
+            active={this.state.display === Display.FRONTEND}
+          >
+            Generate data in frontend
+          </Button>
+
+          <Button
+            variant='info'
+            size='sm'
+            onClick={this.getRemoteData}
+            style={{ margin: 10 }}
+            active={this.state.display === Display.BACKEND}
+          >
+            Generate data in backend
+          </Button>
+
+          <Button
+            variant='info'
+            size='sm'
+            onClick={this.fetchFromStorageData}
+            style={{ margin: 10 }}
+            active={this.state.display === Display.DB}
+          >
+            Fetch data from database
+          </Button>
+        </Row>
+        <Row style={{ marginBottom: 100 }}>
+          <Col className='col-4'>
+            <Chart type='doughnut' data={this.state.chart_data} />
+          </Col>
+          <Col className='col-4'>
+            <Chart type='pie' data={this.state.chart_data} />
+          </Col>
+          <Col className='col-4'>
+            <Chart type='bar' data={this.state.chart_data} />
+          </Col>
+        </Row>
+
+        <Row>
+          <Col className='col-4'>
+            <Chart type='line' data={this.state.chart_data} />
+          </Col>
+          <Col className='col-4'>
+            <Chart type='polarArea' data={this.state.chart_data} />
+          </Col>
+          <Col className='col-4'>
+            <Chart type='horizontalBar' data={this.state.chart_data} />
+          </Col>
+        </Row>
+      </Container>
+    );
+  }
+
+  /* Generate random chart data */
+  generateLocalData() {
+    const newdata = Object.assign({}, this.state.chart_data);
+    newdata.datasets[0].data = [
+      Math.floor(Math.random() * 100 + 1),
+      Math.floor(Math.random() * 100 + 1),
+      Math.floor(Math.random() * 100 + 1),
+    ];
+    this.setState({ display: Display.FRONTEND });
+    this.setState({ chart_data: newdata });
+  }
+
+  /* Generate random chart data from backend*/
+  async getRemoteData() {
+    const newdata = Object.assign({}, this.state.chart_data);
```

Comment:
This is a shallow copy, please use something like [immer.produce](https://github.com/immerjs/immer#reactsetstate-example) in order to create a new value.
Same goes for the other state update functions.